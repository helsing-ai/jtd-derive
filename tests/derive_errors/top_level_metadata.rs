#[derive(jtd_derive::JsonTypedef)]
#[typedef(metadata(foo, bar, foo = 2))]
struct Foo {
    bar: u32,
}

#[derive(jtd_derive::JsonTypedef)]
#[typedef(metadata = "a")]
struct Bar {
    bar: u32,
}

#[derive(jtd_derive::JsonTypedef)]
#[typedef(metadata)]
struct Baz {
    bar: u32,
}

fn main() {}
