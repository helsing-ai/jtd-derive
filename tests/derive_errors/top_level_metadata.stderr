error: expected key-value pair
 --> tests/derive_errors/top_level_metadata.rs:2:20
  |
2 | #[typedef(metadata(foo, bar, foo = 2))]
  |                    ^^^

error: expected key-value pair
 --> tests/derive_errors/top_level_metadata.rs:2:25
  |
2 | #[typedef(metadata(foo, bar, foo = 2))]
  |                         ^^^

error: expected string literal
 --> tests/derive_errors/top_level_metadata.rs:2:36
  |
2 | #[typedef(metadata(foo, bar, foo = 2))]
  |                                    ^

error: the `metadata` parameter must be a list of key-value pairs
 --> tests/derive_errors/top_level_metadata.rs:8:11
  |
8 | #[typedef(metadata = "a")]
  |           ^^^^^^^^^^^^^^

error: the `metadata` parameter must be a list of key-value pairs
  --> tests/derive_errors/top_level_metadata.rs:14:11
   |
14 | #[typedef(metadata)]
   |           ^^^^^^^^
